velox/type/Filter.h

```diff
@@ -70,6 +70,11 @@ class Filter {
 
   virtual std::unique_ptr<Filter> clone() const = 0;
 
+  virtual std::unique_ptr<Filter> clone(bool nullAllowed) const {
```

Comment:
nit: Would it make sense to combine clone() and close(nullAllowed) into a single method?

```
/// Return a copy of this filter. If nullAllowed is set, modified the nullAllowed flag in the copy to match.
virtual std::unique_ptr<Filter> clone(std::optional<bool> nullAllowed = std::nullopt) const
```

It would be nice to add tests for the new method.

Reply:
Done